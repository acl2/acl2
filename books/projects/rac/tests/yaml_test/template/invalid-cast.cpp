#include <ac_int.h>
using namespace std;

// RAC begin

template <int N>
int foo(ac_int<N, false> x) {
  return (int)x;
}

// RAC end
