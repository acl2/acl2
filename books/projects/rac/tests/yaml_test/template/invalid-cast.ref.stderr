invalid-cast.cpp:8 (9-15): Incompatible types: ac_int<N, false> cannot be cast to int
 8 |   return (int)x;
   |          ^^^^^^
note: To convert types depending on template parameters to primitive types, you should add an explicit conversion (like this ac_int<32, true>(...) to convert to int)
